scripts/eval/eval.py

```diff
@@ -131,23 +135,21 @@ def main(cfg: DictConfig):
     models_df = None
     composite_scores = None
     for model_cfg in cfg.models:
-        (in_memory_logger, logger_keys, model_gauntlet_callback, model_gauntlet,
+        (logger, logger_keys, model_gauntlet_callback, model_gauntlet,
          model_gauntlet_df) = evaluate_model(model_cfg, cfg, cfg.run_name,
                                              model_gauntlet_df)
 
         if model_gauntlet_callback is not None:
-            # TODO(bmosaicml) This needs to be refactored to fix the typing issue
-            composite_scores = model_gauntlet_callback.eval_end(
-                None, in_memory_logger)  # type: ignore
+            composite_scores = model_gauntlet_callback.eval_after_all(
```

Comment:
shouldn't the `eval_after_all` just write some data to the `model_gauntlet_callback` (or to a file if you prefer)? Then you don't need to call the event explicitly, as it should just run on the event as intended after eval.